org.eclipse.lsp4e/src/org/eclipse/lsp4e/ui/FoldingPreferencePage.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Vegard IT GmbH and others.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Sebastian Thomschke (Vegard IT GmbH) - initial implementation.
+ *******************************************************************************/
+package org.eclipse.lsp4e.ui;
+
+import org.eclipse.core.runtime.preferences.AbstractPreferenceInitializer;
+import org.eclipse.jface.preference.BooleanFieldEditor;
+import org.eclipse.jface.preference.FieldEditorPreferencePage;
+import org.eclipse.lsp4e.LanguageServerPlugin;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+
+public class FoldingPreferencePage extends FieldEditorPreferencePage implements IWorkbenchPreferencePage {
+
+	public static final String PREF_FOLDING_ENABLED = "foldingReconcilingStrategy.enabled"; //$NON-NLS-1$
+	public static final String PREF_AUTOFOLD_COMMENTS = "foldingReconcilingStrategy.collapseComments"; //$NON-NLS-1$
+	public static final String PREF_AUTOFOLD_REGIONS = "foldingReconcilingStrategy.collapseRegions"; //$NON-NLS-1$
+	public static final String PREF_AUTOFOLD_IMPORT_STATEMENTS = "foldingReconcilingStrategy.collapseImports"; //$NON-NLS-1$
+
+	public static final class PreferenceInitializer extends AbstractPreferenceInitializer {
```

Comment:
Is initializing the preferences to some value the preferred pattern? I think org.eclipse.lsp4e.ui.LoggingPreferencePage does not initialize its preferences, rather it relies on the defaults of IPreferenceStore.getBoolean. I think this is better because one can change the defaults in LSP4E any time and they will be changed for users as well unless they have set the values themselves. Do you agree with that or do you think it is better to initialize the values? Why then?

Reply:
I don't know what the preferred way is TBH. I used it because one pref key needs a default value of true and with the initializer I have a single place to define it.